src/plone/restapi/deserializer/blocks.py

```diff
@@ -12,20 +12,34 @@
 from zope.publisher.interfaces.browser import IBrowserRequest
 
 
-def path2uid(context, path):
+def path2uid(context, portal, href):
     # unrestrictedTraverse requires a string on py3. see:
     # https://github.com/zopefoundation/Zope/issues/674
-    if not isinstance(path, str):
-        path = path.decode("utf-8")
-    obj = context.unrestrictedTraverse(path, None)
+    if not href:
+        return ''
+    portal_url = portal.absolute_url()
+    portal_path = '/'.join(portal.getPhysicalPath())
+    path = href
+    context_url = context.absolute_url()
+    relative_up = len(context_url.split("/")) - len(portal_url.split("/"))
+    if path.startswith(portal_url):
+        path = path[len(portal_url) + 1 :]
+    if not path.startswith(portal_path):
+        path = '{portal_path}/{path}'.format(
+            portal_path=portal_path, path=path.lstrip("/")
+        )
+    obj = portal.unrestrictedTraverse(path, None)
     if obj is None:
-        return None, None
+        return href
     segments = path.split("/")
     suffix = ""
     while not IUUIDAware.providedBy(obj):
         obj = aq_parent(obj)
         suffix += "/" + segments.pop()
-    return IUUID(obj), suffix
```

Comment:
@cekk can you elaborate why we changed the path2uid return from a tuple to returning a full path/href?

Reply:
ok, got it now. you moved logic into the path2uid function...

Reply:
yep ;)